src/swupd/files_test.go

```diff
@@ -0,0 +1,178 @@
+package swupd
+
+import "testing"
+
+func TestSetFileTypeFromFlagFile(t *testing.T) {
+	flags := map[byte]ftype{
```

Comment:
Minor comment: order in maps is not guaranteed to be the same, so it is common to use an array/slice of a structs instead of a map.